src/Main.cpp

```diff
@@ -6,7 +6,8 @@ int main()
     sf::RenderWindow window(sf::VideoMode(800, 600), "Ducky Game!");
     MusicPlayer music;
     Scene scene(window);
-
+    sf::Clock clock;
```

Comment:
okey, i will get to it sometime soon, but I think I'm gonna experiment with global variable instead of passing dt into Update(). changing update function like this would be a pain in the ass considering Martyna's fixes and all ensuing merge conflicts

Reply:
Oh no no no, no global dt please. Passing dt float into update is perfectly normal, every engine known to mankind does that. Even in Unreal you have dt passed in the Tick event

Reply:
Oh and we whould actually have the discussion related to delta here :D

Reply:
Okey, your solution almost works, but not always.
You failed to take one scenario into consideration: someone's pc is too weak to handle the game and doesn't reach the desired frame rate. In that event, because it's the desiredDt that's passed into the Update() and not actual frame duration, if the actual frame rate is lower than expected, the game gets slowed down. After all, the velocity of objects is still multiplied by the default value. 
I know this game isn't very resource demanding and lower than 60fps is nearly impossible to happen, but still, the game should handle that. I believe that passing in actual frame duration, dt, would solve that

Reply:
this is Update(desiredDt) with 30000fps as desiredFrameRate
https://user-images.githubusercontent.com/82574004/169569230-b20c85a9-dc86-43be-88d9-825f713ebdf0.mp4

this is Update(dt) in the same situation
https://user-images.githubusercontent.com/82574004/169569431-f8999119-4d91-48ed-9ce3-97400117968f.mp4

Reply:
Oh, and yeah, I know that there is a while loop that should mean, in the event of a lower frame rate, that the update function is simply called a few times in a row and therefore your solution should work just fine, but, as the evidence above proves, that is not the case. I mean, unless I'm in the wrong again and have misinterpreted your instructions from earlier. Don't eat me, pls